src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadReader.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Spark.Interop.Ipc;
+using Razorvine.Pickle;
+
+namespace Microsoft.Spark.Worker.UnitTest
+{
+    /// <summary>
+    /// Payload reader that reads the output of the inputStream of the socket response  
+    /// </summary>
+    internal sealed class PayloadReader
+    {
+        public static void Read(
```

Comment:
Can we change this to
```C#
public static List<object[]> Read(Stream inputStream)
```
We can assert the states of `timingDataReceived` and `exceptionThrown` inside this function.